topology.graphqls

```diff
@@ -58,12 +63,11 @@ type ServiceInstanceNode {
     serviceId: ID!
     # The literal name of the #serviceId.
     serviceName: String!
-    # The type name may be
-    # 1. The service provider/middleware tech, such as: Tomcat, SpringMVC
-    # 2. Conjectural Service, e.g. MySQL, Redis, Kafka
+
+    # Deprecated since 9.0.0. Instance is an executable unit of service, it doesn't have its own type.
     type: String
-    # It is a conjecture node or real node, to represent an instance.
-    isReal: Boolean!
+    # Deprecated since 9.0.0. Don't need to use this anymore
+    isReal: Boolean
```

Comment:
> Could you share which specific one do you have concern?

Not a critical concern but looking this kind of changes reminded me that if the new backend returns `null` for this field, the previous consumer of this protocol will break, but if the backend keeps returning non-null value then adding the deprecate message should be enough.

Reply:
I think we should return empty string, even the field is changed into `nullable`.

Reply:
> I think we should return empty string, even the field is changed into `nullable`.

OK, that makes sense to me

Reply:
And, the key idea to keep this compatibility to make e2e, CLI, and integration(like HPA) work as usual. But the rocketbot UI is not considered that also working as usual.